### PR TITLE
Add Windows Internet Explorer selenium image

### PR DESCRIPTION
**Pull request by jscotka**
This is a semiautomated script for creating MS Windows selenium hub with Internet Explorer for testing purposes. It uses the readily available VM images from Microsoft which are designed for testing.
    
Regenerating this image requires a human to perform one task, hence the bots will file issues about it but not refresh themselves.
    
This image should not be shared publicly. 
   
 * [x] Integrated into image-create and image-refresh logic
 * [x] Cleaned up messages
 * [x] Added expected IP address for Selenium hub
 * [x] Use CRNL for new lines in scripts
 * [x] image-refresh fedora-24
 * [x] Upload windows-8 image

### CODE REVIEW

**Comment by stefwalter** [comment]
Shouldn't this script be runnable as a non-root user like all our other bootstrap scripts? When I do so I get:

```
Starting install...
ERROR    Network not found: no network with matching name 'default'
Domain installation does not appear to have been successful.
If it was, you can restart your domain by running:
  virsh --connect qemu:///session start win8.1
otherwise, please restart your installation.
Traceback (most recent call last):
  File "test/vm-create", line 191, in <module>
    builder.build()
  File "test/vm-create", line 121, in build
    self.bootstrap_system()
  File "test/vm-create", line 67, in bootstrap_system
    subprocess.check_call([ bootstrap_script, self.machine.image_file ])
  File "/usr/lib64/python2.7/subprocess.py", line 541, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/data/src/cockpit/test/images/scripts/windows-selenium.bootstrap', '/data/src/cockpit/test/tmp/run/windows-selenium.qcow2']' returned non-zero exit status 1
```

**Comment by jscotka** [comment]
Hi, yes, that's true. I've just used as simplest solution as I could. It cannot be fully automated, so that I did not think about that so much. So yes, I can update it and allow to update it by users. (I have to probably use session virt instead of system and use custom cockpit net bridge, instead of default if I inderstand that correctly)

**Comment by stefwalter** [comment]
> Hi, yes, that's true. I've just used as simplest solution as I could. It cannot be fully automated, so that I did not think about that so much. So yes, I can update it and allow to update it by users. (I have to probably use session virt instead of system and use custom cockpit net bridge, instead of default if I inderstand that correctly)

Yes, that would be great. Using the cockpit1 net bridge.

**Comment by jscotka** [comment]
@stefwalter Is it better now? I've replaced using any bridge by using user network, which is accessible to all users.

**Comment by stefwalter** [comment]
Thanks. I got to a different point in the image build:

```
starting converting vmdk to qcow2
qemu-img: Could not open '*.vmdk': Could not open '*.vmdk': No such file or directory
Traceback (most recent call last):
  File "test/vm-create", line 191, in <module>
    builder.build()
  File "test/vm-create", line 121, in build
    self.bootstrap_system()
  File "test/vm-create", line 67, in bootstrap_system
    subprocess.check_call([ bootstrap_script, self.machine.image_file ])
  File "/usr/lib64/python2.7/subprocess.py", line 541, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/data/src/cockpit/test/images/scripts/windows-selenium.bootstrap', '/data/src/cockpit/test/tmp/run/windows-selenium.qcow2']' returned non-zero exit status 1
```

**Comment by jscotka** [comment]
ah, I've tested something and I've had unzipped files that, so that I've didn't see that. this trouble is caused by using bad option for unzip. I'll repair that. Now I'm testing results with new unzip option -o instead of -f

**Comment by stefwalter** [comment]
I've fixed this up, and am uploading the image now.

**Comment by stefwalter** [comment]
image-refresh in progress on falcon.
Log: http://stefw.fedorapeople.org/logs/image-refresh-5477-20170726-103838/log

**Comment by stefwalter** [comment]
image-refresh fedora-24 done: https://github.com/stefwalter/cockpit/commits/image-refresh-fedora-24-20170726-113636

**Comment by stefwalter** [comment]
I've also tested that my changes to the image creation scripts don't break image creation by the bots. However we don't need to commit the resulting image.

File: bots/image-create [hunk 1]
```diff
@@ -142,7 +142,7 @@ class MachineBuilder:
         script = os.path.join(BOTS, "images", "scripts", "%s.setup" % (self.machine.image, ))
 
         if not os.path.exists(script):
-            raise testvm.Failure("Unsupported image %s: %s not found." % (self.machine.image, script))
+            return
```
Comment (dperpeet) [inline bots/image-create]: why should this just quit silently and with success?

File: bots/images/scripts/windows-8.bootstrap [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+sed -i "s/\$/$CR/" *.bat Autorun.inf
+genisoimage -output wininit.iso -volid cidata -joliet *.bat *.exe selenium.jar Autorun.inf
+
+echo "Converting vmdk to qcow2..."
+qemu-img convert -f vmdk -O qcow2 *.vmdk $ORIGIN
+echo "MANUAL part of installation, open graphics console and run D:\\install.bat"
+virt-install -n $VM_NAME -r 4000 --vcpus=4 --os-type=windows --disk $ORIGIN,device=disk --network user -c wininit.iso --noautoconsole --wait=-1 --noreboot
+echo "Installation finished"
+# virsh destroy $VM_NAME
+# virsh undefine $VM_NAME
```
Comment (dperpeet) [inline bots/images/scripts/windows-8.bootstrap]: should we just let these two commands run?

File: bots/image-create (hunk 1 above)
Comment (stefwalter) [inline bots/image-create]: Because some images do not have a setup script.

File: bots/images/scripts/windows-8.bootstrap (hunk 2 above)
Comment (stefwalter) [inline bots/images/scripts/windows-8.bootstrap]: I don't think so. At the current time, there's enough back and forth iteration on this every 30 days, and a human already needs to be involved. I left them commented intentionally.

As it currently stands there's exactly one manual command to be run inside the VM. However after 30 days that may change. I want to make this as painless as possible, hence leaving these commands there.

**Pull request merged**